lib/models/room.js

```diff
@@ -214,43 +292,195 @@ Room.prototype.getAvatarUrl = function(baseUrl, width, height, resizeMethod,
  };
 
 /**
- * Add some events to this room's timeline. Will fire "Room.timeline" for
- * each event added.
+ * Create a new timeline for this room
+ *
+ * @param {Array<MatrixEvent>} stateBeforeEvent state of the room before this event
+ * @return {module:models/event-timeline~EventTimeline} newly-created timeline
+ */
+Room.prototype.createTimeline = function() {
+    if (!this._timelineSupport) {
+        throw Error("timeline support is disabled. Set the 'timelineSupport'" +
+                    " parameter to true when creating MatrixClient to enable" +
+                    " it.");
+    }
+
+    var timeline = new EventTimeline(this.roomId);
+    this._timelines.push(timeline);
```

Comment:
This side-effect is a little surprising given you use the word `create` instead of `add`. Can you clarify in the method docs or rename this method please?